### ISSUE TITLE
Failed prop type: Invalid prop `disabledDates` of type `function` supplied to `YearsGridView`, expected `array`.

### ISSUE THREAD

**Issue by adilusman51** [stephy/CalendarPicker#200]
If disabled dates is provided as a function and user switches to Month/Year grid view, following exception is thrown
**Failed prop type: Invalid prop `disabledDates` of type `function` supplied to `YearsGridView`, expected `array`.**

```javascript
<CalendarPicker
	enableDateChange={true}
	allowRangeSelection={true}
	onDateChange={(date) => console.log(date}
	selectedStartDate={startDate}
	selectedEndDate={endDate}
	disabledDates={(date) => {
		return true;
		}}
	onMonthChange={onMonthChange}
	/>
```

**Comment by adilusman51** [comment]
@peacechen @stephy I recommend that separate props are provided for disabled Months and Years. This would allow individual years and months to be disabled, without the need of decision making based on disabled days of the months/years

**Comment by peacechen** [comment]
Agreed.  I'm working on the scrollable feature which is an extensive change.  I've pushed the first part which is a refactor to make the CalendarDay component self-contained.  A side benefit is that it cleans up the code in the main component.  Would you mind creating a PR to remove the `disabledDates` prop from the Months and Years grid views?  It's currently not used because the original prop was designed for date level granularity, not months or years.

**Comment by adilusman51** [comment]
@peacechen  sure, I can work on this PR

**Comment by peacechen** [comment]
Thanks @adilusman51 

This has been published in 6.1.2 .  npm was having server issues last night and it wouldn't complete until now.
